src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final double MAX_SIZE = 10;
+    private static final int NUMBER_OF_FIGURES = 5;
+
+    public enum Figures {
+        Square,
+        Rectangle,
+        RightTriangle,
+        Circle,
+        IsoscelesTrapezoid
+    }
+
+    public double getRandomSize() {
+        BigDecimal bd;
+        bd = BigDecimal.valueOf(RANDOM.nextDouble(MAX_SIZE) + 1);
+        bd = bd.setScale(1, RoundingMode.HALF_UP);
+        return bd.doubleValue();
+    }
+
+    public Figure getRandomFigure() {
+        int figureType = RANDOM.nextInt(NUMBER_OF_FIGURES);
+        String color;
+
+        switch (figureType) {
+            case 0:
+                double side = getRandomSize();
+                color = ColorSupplier.getRandomColor();
+                return new Square(side, color);
+            case 1:
+                double smallSide = getRandomSize();
+                double bigSide = getRandomSize();
+                color = ColorSupplier.getRandomColor();
+                return new Rectangle(smallSide, bigSide, color);
```

Comment:
same for other figures
```suggestion
                return new Rectangle(getRandomSize(), getRandomSize(), randomColor);
```